user-interface/src/main/java/life/qbic/datamanager/views/projects/project/info/ProjectContentComponent.java

```diff
@@ -29,13 +35,39 @@ public class ProjectContentComponent extends Div {
   private static final long serialVersionUID = -1061134126086910532L;
   private final ProjectDetailsComponent projectDetailsComponent;
 
+  private final ProjectPurchaseService projectPurchaseService;
+  private Context context;
+
   public ProjectContentComponent(
-      @Autowired ProjectDetailsComponent projectDetailsComponent) {
+      @Autowired ProjectDetailsComponent projectDetailsComponent,
+      @Autowired ProjectPurchaseService projectPurchaseService) {
     Objects.requireNonNull(projectDetailsComponent);
+    this.projectPurchaseService = Objects.requireNonNull(projectPurchaseService);
+    UploadPurchaseDialog uploadPurchaseDialog = new UploadPurchaseDialog();
+    uploadPurchaseDialog.setOpened(false);
+    add(uploadPurchaseDialog);
+    Button uploadOffer = new Button("Upload offer");
+    uploadOffer.addClickListener(listener -> uploadPurchaseDialog.open());
+    uploadPurchaseDialog.subscribe(
+        uploadPurchaseDialogConfirmEvent -> {
+          var offers = uploadPurchaseDialog.purchaseItems();
+          uploadPurchaseDialog.close();
```

Comment:
You probably want to close the dialog only after the purchase item is added so that the user can change input if something goes wrong?

Reply:
The saving of offer items is blocking, hence the dialog remains open. This can lead to multiple uploads, if the user hits the save button multiple times.